src/components/NewAppModal.js

```diff
@@ -0,0 +1,161 @@
+              ])
+            ])
+          ])
+        ])
+      ])
+    ])
+  }
+
+  const renderDeleteWarning = () => {
+    return div({ style: { lineHeight: '22px' } }, [
+      div({ style: { marginTop: '0.5rem' } }, [
+        'Deleting your Cloud Environment will also ',
+        span({ style: { fontWeight: 600 } }, ['delete any files on the associated hard disk']),
+        ' (e.g. results files). Double check that your workflow results were written to ',
+        'the data tab in the workspace before clicking “Delete”'
+      ]),
+      div({ style: { marginTop: '1rem' } }, ['Deleting your Cloud Environment will stop your ',
+        'running Galaxy application and your application costs. You can create a new Cloud Environment ',
+        'for Galaxy later, which will take 8-10 minutes.'])
+    ])
+  }
+
+  const renderDefaultCase = () => {
+    return h(Fragment, [
+      div([`Environment ${app ? 'consists' : 'will consist'} of an application and cloud compute.`]),
+      div({ style: { paddingTop: '0.5rem' } }, [
+        div({ style: { ...styles.whiteBoxContainer, marginTop: '0.5rem' } }, [
+          div([
+            div({ style: styles.headerText }, ['Environment Settings']),
+            ul({ style: { paddingLeft: '1rem', lineHeight: 1.5 } }, [
+              li({ style: { marginTop: '1rem' } }, [
+                'Galaxy version xxx'
+              ]),
+              li({ style: { marginTop: '1rem' } }, [
+                'Cloud Compute size of ', span({ style: { fontWeight: 600 } },
+                  [`${getMachineDetails().cpu} CPUS, ${getMachineDetails().memory}  GB of memory, 50 GB disk space`])
+                //TODO: Define the disk space using DEFAULT_DISK_SIZE from the mob_pd branch
+              ]),
+              li({ style: { marginTop: '1rem' } }, [
+                'Running cloud compute costs ',
+                span({ style: { fontWeight: 600 } }, '$0.00 per hr')
+                //TODO: Calculate cost
+              ])
+            ]),
+            h(Link, {
+              ...Utils.newTabLinkProps,
+              // TODO: Get the link from comms for this
+              href: ''
+            }, ['Learn more about Galaxy interactive environments.'])
+          ])
+        ])
+      ])
+    ])
+  }
+
+  const contents = Utils.switchCase(viewMode,
+    ['createWarn', renderCreateWarning],
+    ['deleteWarn', renderDeleteWarning],
+    [Utils.DEFAULT, renderDefaultCase]
+  )
+
+  return div({ style: styles.drawerContent }, [
+    h(TitleBar, {
+      title: Utils.switchCase(viewMode,
+        ['deleteWarn', () => 'Delete Cloud Environment for Galaxy'],
+        [Utils.DEFAULT, () => 'Cloud environment']
+      ),
+      onDismiss,
+      onPrevious: !!viewMode ? () => setViewMode(undefined) : undefined,
+      style: { margin: '2rem 0 0 1.5rem' }
+    }),
+    div({ style: { padding: '0.5rem 1.5rem 1.5rem', flexGrow: 1, display: 'flex', flexDirection: 'column' } }, [contents, renderBottomButtons()])
```

Comment:
nit: This wrapper and the `TitleBar` are just adding extra padding inside of what's being provided by `styles.drawerContent`. We could simplify the layout by combining the padding.